<div class="body" role="main">
<section id="modeller">
<h1>Modeller<a class="headerlink" href="#modeller" title="Permalink to this headline">Â¶</a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">openmm.app.modeller.</span></span><span class="sig-name descname"><span class="pre">Modeller</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">topology</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">positions</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Modeller provides tools for editing molecular models, such as adding water or missing hydrogens.</p>
<p>To use it, create a Modeller object, specifying the initial Topology and atom positions.  You can
then call various methods to change the model in different ways.  Each time you do, a new Topology
and list of coordinates is created to represent the changed model.  Finally, call getTopology()
and getPositions() to get the results.</p>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">topology</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">positions</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.__init__" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Create a new Modeller object</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>topology</strong> (<a class="reference internal" href="openmm.app.topology.Topology.html#openmm.app.topology.Topology" title="openmm.app.topology.Topology"><em>Topology</em></a>) â the initial Topology of the model</p></li>
<li><p><strong>positions</strong> (<em>list</em>) â the initial atomic positions</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%"/>
<col style="width: 90%"/>
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.__init__" title="openmm.app.modeller.Modeller.__init__"><code class="xref py py-obj docutils literal notranslate"><span class="pre">__init__</span></code></a>(topology,Â positions)</p></td>
<td><p>Create a new Modeller object</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.add" title="openmm.app.modeller.Modeller.add"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add</span></code></a>(addTopology,Â addPositions)</p></td>
<td><p>Add chains, residues, atoms, and bonds to the model.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.addExtraParticles" title="openmm.app.modeller.Modeller.addExtraParticles"><code class="xref py py-obj docutils literal notranslate"><span class="pre">addExtraParticles</span></code></a>(forcefield[,Â â¦])</p></td>
<td><p>Add missing extra particles to the model that are required by a force field.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.addHydrogens" title="openmm.app.modeller.Modeller.addHydrogens"><code class="xref py py-obj docutils literal notranslate"><span class="pre">addHydrogens</span></code></a>([forcefield,Â pH,Â variants,Â â¦])</p></td>
<td><p>Add missing hydrogens to the model.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.addMembrane" title="openmm.app.modeller.Modeller.addMembrane"><code class="xref py py-obj docutils literal notranslate"><span class="pre">addMembrane</span></code></a>(forcefield[,Â lipidType,Â â¦])</p></td>
<td><p>Add a lipid membrane to the model.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.addSolvent" title="openmm.app.modeller.Modeller.addSolvent"><code class="xref py py-obj docutils literal notranslate"><span class="pre">addSolvent</span></code></a>(forcefield[,Â model,Â boxSize,Â â¦])</p></td>
<td><p>Add solvent (both water and ions) to the model to fill a periodic box.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.convertWater" title="openmm.app.modeller.Modeller.convertWater"><code class="xref py py-obj docutils literal notranslate"><span class="pre">convertWater</span></code></a>([model])</p></td>
<td><p>Convert all water molecules to a different water model.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.delete" title="openmm.app.modeller.Modeller.delete"><code class="xref py py-obj docutils literal notranslate"><span class="pre">delete</span></code></a>(toDelete)</p></td>
<td><p>Delete chains, residues, atoms, and bonds from the model.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.deleteWater" title="openmm.app.modeller.Modeller.deleteWater"><code class="xref py py-obj docutils literal notranslate"><span class="pre">deleteWater</span></code></a>()</p></td>
<td><p>Delete all water molecules from the model.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.getPositions" title="openmm.app.modeller.Modeller.getPositions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">getPositions</span></code></a>()</p></td>
<td><p>Get the atomic positions.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.getTopology" title="openmm.app.modeller.Modeller.getTopology"><code class="xref py py-obj docutils literal notranslate"><span class="pre">getTopology</span></code></a>()</p></td>
<td><p>Get the Topology of the model.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#openmm.app.modeller.Modeller.loadHydrogenDefinitions" title="openmm.app.modeller.Modeller.loadHydrogenDefinitions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">loadHydrogenDefinitions</span></code></a>(file)</p></td>
<td><p>Load an XML file containing definitions of hydrogens that should be added by addHydrogens().</p></td>
</tr>
</tbody>
</table>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.getTopology">
<span class="sig-name descname"><span class="pre">getTopology</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.getTopology" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Get the Topology of the model.</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.getPositions">
<span class="sig-name descname"><span class="pre">getPositions</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.getPositions" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Get the atomic positions.</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.add">
<span class="sig-name descname"><span class="pre">add</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">addTopology</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">addPositions</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.add" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Add chains, residues, atoms, and bonds to the model.</p>
<p>Specify what to add by providing a new Topology object and the
corresponding atomic positions. All chains, residues, atoms, and bonds
contained in the Topology are added to the model.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>addTopology</strong> (<a class="reference internal" href="openmm.app.topology.Topology.html#openmm.app.topology.Topology" title="openmm.app.topology.Topology"><em>Topology</em></a>) â a Topology whose contents should be added to the model</p></li>
<li><p><strong>addPositions</strong> (<em>list</em>) â the positions of the atoms to add</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.delete">
<span class="sig-name descname"><span class="pre">delete</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">toDelete</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.delete" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Delete chains, residues, atoms, and bonds from the model.</p>
<p>You can specify objects to delete at any granularity: atoms, residues, or chains.  Passing
in an Atom object causes that Atom to be deleted.  Passing in a Residue object causes that
Residue and all Atoms it contains to be deleted.  Passing in a Chain object causes that
Chain and all Residues and Atoms it contains to be deleted.</p>
<p>In all cases, when an Atom is deleted, any bonds it participates in are also deleted.
You also can specify a bond (as a tuple of Atom objects) to delete just that bond without
deleting the Atoms it connects.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>toDelete</strong> (<em>list</em>) â a list of Atoms, Residues, Chains, and bonds (specified as tuples of
Atoms) to delete</p>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.deleteWater">
<span class="sig-name descname"><span class="pre">deleteWater</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.deleteWater" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Delete all water molecules from the model.</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.convertWater">
<span class="sig-name descname"><span class="pre">convertWater</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'tip3p'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.convertWater" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Convert all water molecules to a different water model.</p>
<div class="admonition-deprecated admonition">
<p class="admonition-title">Deprecated</p>
<blockquote>
<div><p>Use addExtraParticles() instead.  It performs the same function but in a more general way.</p>
</div></blockquote>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>model</strong> (<em>string='tip3p'</em>) â the water model to convert to.  Supported values are âtip3pâ,
âspceâ, âtip4pewâ, and âtip5pâ.</p>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.addSolvent">
<span class="sig-name descname"><span class="pre">addSolvent</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">forcefield</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'tip3p'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boxSize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boxVectors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">padding</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">numAdded</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boxShape</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'cube'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">positiveIon</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Na+'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negativeIon</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Cl-'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ionicStrength</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">Quantity(value=0,</span> <span class="pre">unit=molar)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neutralize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residueTemplates</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.addSolvent" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Add solvent (both water and ions) to the model to fill a periodic box.</p>
<p>The algorithm works as follows:</p>
<ol class="arabic simple">
<li><p>Water molecules are added to fill the box.</p></li>
<li><p>Water molecules are removed if their distance to any solute atom is less than the sum of their van der Waals radii.</p></li>
<li><p>If the solute is charged and neutralize=True, enough positive or negative ions are added to neutralize it.  Each ion is added by
randomly selecting a water molecule and replacing it with the ion.</p></li>
<li><p>Ion pairs are added to give the requested total ionic strength.  Note that only monovalent ions are currently supported.</p></li>
</ol>
<p>The box size can be specified in any of several ways:</p>
<ol class="arabic simple">
<li><p>You can explicitly give the vectors defining the periodic box to use.</p></li>
<li><p>Alternatively, for a rectangular box you can simply give the dimensions of the unit cell.</p></li>
<li><p>You can give a padding distance.  A bounding sphere containing the solute is determined, and the box size is
set to (sphere diameter)+(padding).  This guarantees no atom in the solute will come closer than the padding
distance to any atom of another periodic copy.  If the sphere diameter is less than the padding distance,
the box size is set to 2*(padding) to ensure no atom is closer than the padding distance to two periodic
copies of any other atom.</p></li>
<li><p>You can specify the total number of molecules (both waters and ions) to add.  A box is then created whose size is
just large enough hold the specified amount of solvent.</p></li>
<li><p>Finally, if none of the above options is specified, the existing Topologyâs box vectors are used.</p></li>
</ol>
<p>When specifying either a padding distance or a number of molecules, you can specify a shape for the periodic box:
cubic, rhombic dodecahedron, or truncated octahedron.  Using a non-rectangular box allows the same distance
between periodic copies to be achieved with a smaller box.  The most compact option is a rhombic dodecahedron,
for which the box volume is 70.7% the volume of a cubic box with the same amount of padding.</p>
<p>There exist many different water models, many of which are very similar to each other. This method creates
preequilibrated water boxes for a limited set of water models. In most cases, they work equally well for other models
that involve the same number of particles. For example, to simulate a box of TIP3P-FB water, use this method to
create a box of TIP3P water, construct a System using TIP3P-FB parameters, and perform a local energy minimization
to correct for the small differences between the models. Likewise, a box of TIP4P-Ew water can be used for most
four site water models.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>forcefield</strong> (<a class="reference internal" href="openmm.app.forcefield.ForceField.html#openmm.app.forcefield.ForceField" title="openmm.app.forcefield.ForceField"><em>ForceField</em></a>) â the ForceField to use for determining van der Waals radii and atomic charges</p></li>
<li><p><strong>model</strong> (<em>str='tip3p'</em>) â the water model to use.  Supported values are âtip3pâ, âspceâ, âtip4pewâ, âtip5pâ, and âswm4ndpâ (polarizable).</p></li>
<li><p><strong>boxSize</strong> (<em>Vec3=None</em>) â the size of the box to fill with water</p></li>
<li><p><strong>boxVectors</strong> (<em>tuple of Vec3=None</em>) â the vectors defining the periodic box to fill with water</p></li>
<li><p><strong>padding</strong> (<em>distance=None</em>) â the padding distance to use</p></li>
<li><p><strong>numAdded</strong> (<em>int=None</em>) â the total number of molecules (waters and ions) to add</p></li>
<li><p><strong>boxShape</strong> (<em>str='cube'</em>) â the box shape to use.  Allowed values are âcubeâ, âdodecahedronâ, and âoctahedronâ.  If padding and numAdded
are both None, this is ignored.</p></li>
<li><p><strong>positiveIon</strong> (<em>string='Na+'</em>) â the type of positive ion to add.  Allowed values are âCs+â, âK+â, âLi+â, âNa+â, and âRb+â</p></li>
<li><p><strong>negativeIon</strong> (<em>string='Cl-'</em>) â the type of negative ion to add.  Allowed values are âCl-â, âBr-â, âF-â, and âI-â. Be aware
that not all force fields support all ion types.</p></li>
<li><p><strong>ionicStrength</strong> (<em>concentration=0*molar</em>) â the total concentration of ions (both positive and negative) to add.  This
does not include ions that are added to neutralize the system.
Note that only monovalent ions are currently supported.</p></li>
<li><p><strong>neutralize</strong> (<em>bool=True</em>) â whether to add ions to neutralize the system</p></li>
<li><p><strong>residueTemplates</strong> (<em>dict=dict</em><em>(</em><em>)</em>) â specifies which template the ForceField should use for particular residues.  The keys
should be Residue objects from the Topology, and the values should be the names of the
templates to use for them.  This is useful when a ForceField contains multiple templates
that can match the same residue (e.g Fe2+ and Fe3+ templates in the ForceField for a
monoatomic iron ion in the Topology).</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.loadHydrogenDefinitions">
<em class="property"><span class="pre">static</span> </em><span class="sig-name descname"><span class="pre">loadHydrogenDefinitions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.loadHydrogenDefinitions" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Load an XML file containing definitions of hydrogens that should be added by addHydrogens().</p>
<p>The built in hydrogens.xml file containing definitions for standard amino acids and nucleotides is loaded automatically.
This method can be used to load additional definitions for other residue types.  They will then be used in subsequent
calls to addHydrogens().</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>file</strong> (<em>string</em><em> or </em><em>file</em>) â An XML file containing hydrogen definitions.  It may be either an
absolute file path, a path relative to the current working
directory, a path relative to this moduleâs data subdirectory (for
built in sets of definitions), or an open file-like object with a read()
method from which the data can be loaded.</p>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.addHydrogens">
<span class="sig-name descname"><span class="pre">addHydrogens</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">forcefield</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pH</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variants</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">platform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residueTemplates</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.addHydrogens" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Add missing hydrogens to the model.</p>
<p>Some residues can exist in multiple forms depending on the pH and properties of the local environment.  These
variants differ in the presence or absence of particular hydrogens.  In particular, the following variants
are supported:</p>
<dl class="simple">
<dt>Aspartic acid:</dt><dd><p>ASH: Neutral form with a hydrogen on one of the delta oxygens
ASP: Negatively charged form without a hydrogen on either delta oxygen</p>
</dd>
<dt>Cysteine:</dt><dd><p>CYS: Neutral form with a hydrogen on the sulfur
CYX: No hydrogen on the sulfur (either negatively charged, or part of a disulfide bond)</p>
</dd>
<dt>Glutamic acid:</dt><dd><p>GLH: Neutral form with a hydrogen on one of the epsilon oxygens
GLU: Negatively charged form without a hydrogen on either epsilon oxygen</p>
</dd>
<dt>Histidine:</dt><dd><p>HID: Neutral form with a hydrogen on the ND1 atom
HIE: Neutral form with a hydrogen on the NE2 atom
HIP: Positively charged form with hydrogens on both ND1 and NE2
HIN: Negatively charged form without a hydrogen on either ND1 or NE2</p>
</dd>
<dt>Lysine:</dt><dd><p>LYN: Neutral form with two hydrogens on the zeta nitrogen
LYS: Positively charged form with three hydrogens on the zeta nitrogen</p>
</dd>
</dl>
<p>The variant to use for each residue is determined by the following rules:</p>
<ol class="arabic simple">
<li><p>The most common variant at the specified pH is selected.</p></li>
<li><p>Any Cysteine that participates in a disulfide bond uses the CYX variant regardless of pH.</p></li>
<li><p>For a neutral Histidine residue, the HID or HIE variant is selected based on which one forms a better hydrogen bond.</p></li>
</ol>
<p>You can override these rules by explicitly specifying a variant for any residue.  To do that, provide a list for the
âvariantsâ parameter, and set the corresponding element to the name of the variant to use.</p>
<p>A special case is when the model already contains a hydrogen that should not be present in the desired variant.
If you explicitly specify a variant using the âvariantsâ parameter, the residue will be modified to match the
desired variant, removing hydrogens if necessary.  On the other hand, for residues whose variant is selected
automatically, this function will only add hydrogens.  It will never remove ones that are already present in the
model, regardless of the specified pH.</p>
<p>In all cases, the positions of existing atoms (including existing hydrogens) are not modified.</p>
<p>Definitions for standard amino acids and nucleotides are built in.  You can call loadHydrogenDefinitions() to load
additional definitions for other residue types.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>forcefield</strong> (<em>ForceField=None</em>) â the ForceField to use for determining the positions of hydrogens.
If this is None, positions will be picked which are generally
reasonable but not optimized for any particular ForceField.</p></li>
<li><p><strong>pH</strong> (<em>float=7.0</em>) â the pH based on which to select variants</p></li>
<li><p><strong>variants</strong> (<em>list=None</em>) â an optional list of variants to use.  If this is specified, its
length must equal the number of residues in the model.  variants[i]
is the name of the variant to use for residue i (indexed starting at
0). If an element is None, the standard rules will be followed to
select a variant for that residue.</p></li>
<li><p><strong>platform</strong> (<em>Platform=None</em>) â the Platform to use when computing the hydrogen atom positions.  If
this is None, the default Platform will be used.</p></li>
<li><p><strong>residueTemplates</strong> (<em>dict=dict</em><em>(</em><em>)</em>) â specifies which template the ForceField should use for particular residues.  The keys
should be Residue objects from the Topology, and the values should be the names of the
templates to use for them.  This is useful when a ForceField contains multiple templates
that can match the same residue (e.g Fe2+ and Fe3+ templates in the ForceField for a
monoatomic iron ion in the Topology).</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>a list of what variant was actually selected for each residue,
in the same format as the variants parameter</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.addExtraParticles">
<span class="sig-name descname"><span class="pre">addExtraParticles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">forcefield</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ignoreExternalBonds</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residueTemplates</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.addExtraParticles" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Add missing extra particles to the model that are required by a force
field.</p>
<p>Some force fields use âextra particlesâ that do not represent
actual atoms, but still need to be included in the System.  Examples
include lone pairs, Drude particles, and the virtual sites used in some
water models to adjust the charge distribution.  Extra particles can be
recognized by the fact that their element is None.</p>
<p>This method is primarily used to add extra particles, but it can also
remove them.  It tries to match every residue in the Topology to a
template in the force field.  If there is no match, it will both add
and remove extra particles as necessary to make it match.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>forcefield</strong> (<a class="reference internal" href="openmm.app.forcefield.ForceField.html#openmm.app.forcefield.ForceField" title="openmm.app.forcefield.ForceField"><em>ForceField</em></a>) â the ForceField defining what extra particles should be present</p></li>
<li><p><strong>ignoreExternalBonds</strong> (<em>boolean=False</em>) â If true, ignore external bonds when matching residues to templates.
This is useful when the Topology represents one piece of a larger
molecule, so chains are not terminated properly.</p></li>
<li><p><strong>residueTemplates</strong> (<em>dict=dict</em><em>(</em><em>)</em>) â specifies which template the ForceField should use for particular residues.  The keys
should be Residue objects from the Topology, and the values should be the names of the
templates to use for them.  This is useful when a ForceField contains multiple templates
that can match the same residue (e.g Fe2+ and Fe3+ templates in the ForceField for a
monoatomic iron ion in the Topology).</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="openmm.app.modeller.Modeller.addMembrane">
<span class="sig-name descname"><span class="pre">addMembrane</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">forcefield</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lipidType</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'POPC'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">membraneCenterZ</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">Quantity(value=0,</span> <span class="pre">unit=nanometer)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minimumPadding</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">Quantity(value=1,</span> <span class="pre">unit=nanometer)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">positiveIon</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Na+'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negativeIon</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Cl-'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ionicStrength</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">Quantity(value=0,</span> <span class="pre">unit=molar)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neutralize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residueTemplates</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#openmm.app.modeller.Modeller.addMembrane" title="Permalink to this definition">Â¶</a></dt>
<dd><p>Add a lipid membrane to the model.</p>
<p>This method actually adds both a membrane and a water box.  It is best to build them together,
both to avoid adding waters inside the membrane and to ensure that lipid head groups are properly
solvated.  For that reason, this method includes many of the same arguments as addSolvent().</p>
<p>The membrane is added in the XY plane, and the existing protein is assumed to already be oriented
and positioned correctly.  When possible, it is recommended to start with a model from the
Orientations of Proteins in Membranes (OPM) database at <a class="reference external" href="http://opm.phar.umich.edu">http://opm.phar.umich.edu</a>.  Otherwise, it
is up to you to select the protein position yourself.</p>
<p>The algorithm is based on the one described in Wolf et al., J. Comp. Chem. 31, pp. 2169-2174 (2010).
It begins by tiling copies of a pre-equilibrated membrane patch to create a membrane of the desired
size.  Next it scales down the protein by 50% along the X and Y axes.  Any lipid within a cutoff
distance of the scaled protein is removed.  It also ensures that equal numbers of lipids are removed
from each leaf of the membrane.  Finally, it runs molecular dynamics to let the membrane relax while
gradually scaling the protein back up to its original size.</p>
<p>The size of the membrane and water box are determined by the minimumPadding argument.  All
pre-existing atoms are guaranteed to be at least this far from any edge of the periodic box.  It
is also possible for the periodic box to have more padding than requested.  In particular, it only
adds whole copies of the pre-equilibrated membrane patch, so the box dimensions will always be
integer multiples of the patch size.  That may lead to a larger membrane than what you requested.</p>
<p>This method has built in support for POPC, POPE, DLPC, DLPE, DMPC, DOPC and DPPC lipids.
You can also build other types of membranes by providing a pre-equilibrated, solvated membrane patch
that can be tiled in the XY plane to form the membrane.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>forcefield</strong> (<a class="reference internal" href="openmm.app.forcefield.ForceField.html#openmm.app.forcefield.ForceField" title="openmm.app.forcefield.ForceField"><em>ForceField</em></a>) â the ForceField to use for determining atomic charges and for relaxing the membrane</p></li>
<li><p><strong>lipidType</strong> (<em>string</em><em> or </em><em>object</em>) â the type of lipid to use.  Supported string values are âPOPCâ, âPOPEâ, âDLPCâ, âDLPEâ, âDMPCâ,
âDOPCâ, and âDPPCâ.  For other types of lipids, provide a PDBFile or PDBxFile object (or any
other object with âtopologyâ and âpositionsâ fields) containing a membrane patch.</p></li>
<li><p><strong>membraneCenterZ</strong> (<em>distance=0*nanometer</em>) â the position along the Z axis of the center of the membrane</p></li>
<li><p><strong>minimumPadding</strong> (<em>distance=1*nanometer</em>) â the padding distance to use</p></li>
<li><p><strong>positiveIon</strong> (<em>string='Na+'</em>) â the type of positive ion to add.  Allowed values are âCs+â, âK+â, âLi+â, âNa+â, and âRb+â</p></li>
<li><p><strong>negativeIon</strong> (<em>string='Cl-'</em>) â the type of negative ion to add.  Allowed values are âCl-â, âBr-â, âF-â, and âI-â. Be aware
that not all force fields support all ion types.</p></li>
<li><p><strong>ionicStrength</strong> (<em>concentration=0*molar</em>) â the total concentration of ions (both positive and negative) to add.  This
does not include ions that are added to neutralize the system.
Note that only monovalent ions are currently supported.</p></li>
<li><p><strong>neutralize</strong> (<em>bool=True</em>) â whether to add ions to neutralize the system</p></li>
<li><p><strong>residueTemplates</strong> (<em>dict=dict</em><em>(</em><em>)</em>) â specifies which template the ForceField should use for particular residues.  The keys
should be Residue objects from the Topology, and the values should be the names of the
templates to use for them.  This is useful when a ForceField contains multiple templates
that can match the same residue (e.g Fe2+ and Fe3+ templates in the ForceField for a
monoatomic iron ion in the Topology).</p></li>
</ul>
</dd>
</dl>
</dd></dl>
</dd></dl>
</section>
</div>